app/src/main/java/com/iti/android/team1/ebuy/ui/login_screen/viewmodel/LoginScreenViewModel.kt

```diff
@@ -13,23 +13,46 @@ import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.launch
 
+
 class LoginScreenViewModel(private val repository: IRepository) : ViewModel() {
 
     private val _loginState: MutableStateFlow<ResultState<Customer>> =
         MutableStateFlow(ResultState.Loading)
     val loginState get() = _loginState.asStateFlow()
 
-    fun makeLoginRequest(customerLogin: CustomerLogin) {
+    fun makeLoginRequest(email: String, password: String) {
+
+        val customerLogin = CustomerLogin(
+            email = email,
+            password = repository.encodePassword(password)
+        )
```

Comment:
we put the encoding in repository not in viewmodel

Reply:
ill update it